src/Analysis/Ast/Impl/Modules/Resolution/ModuleResolutionBase.cs

```diff
@@ -77,8 +77,16 @@ internal abstract class ModuleResolutionBase {
             ).Select(mp => mp.ModuleName).Where(n => !string.IsNullOrEmpty(n)).TakeWhile(_ => !cancellationToken.IsCancellationRequested).ToList();
         }
 
-        public IPythonModule GetImportedModule(string name)
-            => _modules.TryGetValue(name, out var module) ? module : null;
+        public IPythonModule GetImportedModule(string name) {
+            var module = _interpreter.ModuleResolution.GetSpecializedModule(name);
+            if (module != null) {
+                return module;
+            }
+            return _modules.TryGetValue(name, out module) ? module : null;
+        }
+
+        public void SetUserSearchPaths(in IEnumerable<string> searchPaths)
```

Comment:
With this added to the interface, MainModuleResolution can now probably call this instead of going to the path resolver directly.